packages/material-ui/src/Grid/Grid.js

```diff
@@ -121,14 +141,14 @@ export const styles = (theme) => ({
   'direction-xs-column': {
     flexDirection: 'column',
     '& > $item': {
-      maxWidth: 'none',
+      maxWidth: 'none !important',
```

Comment:
Is this absolutely necessary? A component library using `!important` can be very frustrating unless we effectively want to disable customizeability.

Reply:
@eps1lon  I don't think so, I saw issue #23891 when that code was added, but I didn't actually understand the problem that needed to be solved, so I just presumed that you wanted to keep that.

Reply:
>  I saw issue #23891 when that code was added, but I didn't actually understand the problem that needed to be solved, so I just presumed that you wanted to keep that.

The PR that closed that issue did not add `!important`. Did you not add `!important` in this PR i.e. from which location did you copy it?

Reply:
I added `!important` because I presumed was necessary because of that issue if is not you can remove it.

Reply:
It also changes existing demos (https://www.argos-ci.com/mui-org/material-ui/builds/6823) which is also not explained in this PR. Is this demo expected to be backwards incompatible or are the changes intended?

For maintenance it's very important that changes are summarized. Otherwise we might revert this change in the future assuming we misidentified the issue. Right now it only links to to 20+ comment issue from which it's not trivial to infer what the problem is.

It seems to me that this should be doable with a codesandbox, the expected behavior and the actual behavior. We can then compare the codesandbox with the currently published code and the code introduced here.

Reply:
I have a [Demo](https://codesandbox.io/s/practical-elion-ifsrq?file=/src/App.js) ready here. About the changes to the Auto layout demo, it might be a good thing, because when the dev sets the spacing, I think he thinks that will only be applied internally, in the current code it adds margin-bottom due to the padding-bottom of the Grid item.

Reply:
You can write a single codesandbox where we can just change the dependency to the package built from this package. It's a bit hard to grok the demo with all this unused  code.

Reply:
In your demo no `!important` is used as far as I can tell. Does this mean we can remove it from the proposed solution considering the style is applied regardless of `!important`?

Reply:
This is the difference, as I said I don't see a reason to use this `max-width: none`. Because I'm using `xs={6}` inside a `xs={6}` column-reverse, I think that should be the experience.
![image](https://user-images.githubusercontent.com/30958574/104203000-62dd2b80-540a-11eb-8712-bb92a6cb11ff.png)

Reply:
I have updated the demos so the visual regression tests are correct. They are positive side effects of the fix. For instance, the change on the AutoGrid.tsx or List demonstrates that the fix is working. The changes in the TransferList are related to CSS that shouldn't have been here in the first place (noop).

Reply:
Regarding the `!important` style, #23891 is the tip of the iceberg. The problem was reported long before in #13277. I personally can't think of an alternative but maybe @garth has ideas on how to work around it.

Reply:
> Regarding the !important style, #23891 is the tip of the iceberg. The problem was reported long before in #13277.

Both of these are closed. Why do you think this change relates to these issues?